torch/csrc/autograd/saved_variable.cpp

```diff
@@ -20,7 +20,7 @@ SavedVariable::SavedVariable(const Variable& variable, bool is_output) {
     has_grad_fn_ = !variable.is_leaf();
     // These copies are all shared_ptr copies, so slightly more expensive.
     // Do them here instead of in the init list in case data is undefined.
-    data_ = variable.data();
+    data_ = at::Tensor(variable.unsafeGetTensorImpl()->shallow_copy_and_detach());
```

Comment:
If we were to do `data_ = variable` here, it would create a reference cycle for an in-place operation that saves its output (such as relu/threshold): `base.grad_fn.fn.input_.base`. The right way to do it is to save a shallow copy of the variable, which is what we do here.